packages/venia-ui/lib/components/Checkout/form.js

```diff
@@ -1,21 +1,64 @@
-import React, { useState } from 'react';
-import { shape, string } from 'prop-types';
+import React, { useCallback, useState } from 'react';
+import { func, shape, string } from 'prop-types';
+import { AlertCircle as AlertCircleIcon } from 'react-feather';
 
-import { mergeClasses } from '../../classify';
+import { useToasts } from '@magento/peregrine';
+import { useForm } from '@magento/peregrine/lib/talons/Checkout/useForm';
 
+import { mergeClasses } from '../../classify';
+import GET_ALL_COUNTRIES from '../../queries/getAllCountries.graphql';
+import Icon from '../Icon';
+import LoadingIndicator from '../LoadingIndicator';
 import EditableForm from './editableForm';
 import Overview from './overview';
 import defaultClasses from './form.css';
 
+const ErrorIcon = <Icon src={AlertCircleIcon} attrs={{ width: 18 }} />;
+const loadingIndicator = (
+    <LoadingIndicator>{`Loading Checkout...`}</LoadingIndicator>
+);
+
 /**
  * The Form component is similar to Flow in that it renders either the overview
  * or the editable form based on the 'editing' state value.
  */
 const Form = props => {
-    const classes = mergeClasses(defaultClasses, props.classes);
+    const { setStep } = props;
+
+    const [, { addToast }] = useToasts();
     const [editing, setEditing] = useState(null);
+    const [didShowToast, setDidShowToast] = useState(false);
+
+    const talonProps = useForm({ countriesQuery: GET_ALL_COUNTRIES });
+    const { countries, countriesError, isLoadingCountries } = talonProps;
+
+    const handleCountriesError = useCallback(() => {
+        if (!didShowToast) {
+            addToast({
+                type: 'error',
+                icon: ErrorIcon,
+                message: 'Unable to load checkout. Please try again later.',
+                timeout: 3000
+            });
+
+            setDidShowToast(true);
```

Comment:
Rather than a callback you should be able to do this with an effect based on `hasErrors`.

Reply:
Thank you! https://github.com/magento/pwa-studio/pull/1993/commits/8ff820c2f73c711c8210e3b53b7ddaa287cbdbde.